pages/home/home.ejs

```diff
@@ -2,108 +2,512 @@
 <html>
   <head>
     <%- include('../partials/head'); %>
+    <link href="/stylesheets/landing.css" rel="stylesheet">
   </head>
-  <body>
-    <%- include('../partials/navbar', {navPage: 'home'}); %>
-    <div id="content" class="container">
 
-      <% if (devMode) { %>
-      <div class="card mb-4">
-        <div class="card-header bg-primary text-white">Development Mode</div>
-        <div class="card-body">
-          <p>
-            PrairieLearn is running in Development Mode. Click
-            the <strong>“Load from disk”</strong> button above to load
-            question and assessment definitions from JSON files on
-            disk.
-          </p>
-          <p>
-            You need to click “Load from disk” every time that a JSON
-            file is changed on disk. Changes to other files (JS, HTML,
-            etc) will be automatically loaded every time you navigate
-            to a different page or if you reload the current page in
-            your web browser.
-          </p>
-          <p class="mb-0">
-            See
-            the <a href="https://github.com/PrairieLearn/PrairieLearn/blob/master/README.md">PrairieLearn
-            documentation</a> for information on creating questions
-            and assessments.
-          </p>
-        </div>
-      </div>
-      <% } %>
+  <%
+    var isAuthenticated = locals.authn_user;
+    var isAdministrator = locals.is_administrator;
+    var isStudent = typeof locals.authz_data !== "undefined" && locals.authz_data.role === 'Student';
+    var isInstructor = typeof locals.authz_data !== "undefined" && locals.authz_data.role === 'Instructor';
+  %>
```

Comment:
I'd rather set these variables in `home.js`.

For coercing variables to booleans a common pattern is `locals.isAuthenticated = !!locals.authn_user;`

The variables `isStudent` and `isInstructor` are not valid on the home/landing page, because they are specific to particular courses. A user can be an instructor in one course but a student in another.

I think the right strategy is to make the homepage always applicable to all users and don't rely on any flags (like `isAdministrator`, `isInstructor`, `isStudent`). I think that `isAuthenticated` is the only variable that we should be using here.

Reply:
@mwest1066 refactoring should be done